### PR TITLE
[BOP-1023] Add inline and inline compact options to the select kit for inline usage

### PR DESCRIPTION
**Pull request by haydenrou**
#### Screens

Screenshot 2021-07-28 at 11 29 18 AM
Screenshot 2021-07-28 at 11 29 23 AM

#### Breaking Changes

None

#### Runway Ticket URL

https://nitro.powerhrg.com/runway/backlog_items/BOP-1023

#### How to test this

Test from the Select Kit page, according to [this](https://pr234.playgrounds.beta.hq.powerapp.cloud/) prototype, it should be usable for both the Story/Bug change select and also the Sprint change select.

#### Checklist:

- [ ] **LABELS** Add a label: `enhancement`, `bug`, `improvement`, `new kit`, `deprecated`, or `breaking`. See [Changelog & Labels](https://github.com/powerhome/playbook/wiki/Changelog-&-Labels) for details.
- [ ] **URGENCY** Please select a release milestone
- [ ] **DEPLOY** Please add the `Milano` label when you are ready for a review.
- [ ] **SCREENSHOT** Please add a screen shot or two.
- [ ] **SPECS** Please cover your changes with specs.
- [ ] **READ DOCS** Please make sure you have read and understand the [Playbook Release Process](https://github.com/powerhome/playbook/wiki/Playbook-Releases)

*The normal release cut off deadline is 3p EDT each week. Please reach out to the release team if you have an urgent request or need a release off cycle.*

### CODE REVIEW

**Comment by evalouderback** [comment]
This looks good so far @haydenrou It looks like the icon you're using for the arrow is `chevron-down` and not `angle-down` (which is slightly smaller). Do we want the kit to force uppercase on the compact version @jasoncypret ?

File: playbook/app/pb_kits/playbook/pb_select/select.html.erb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <%= content_tag(:div,
   data: object.data,
   aria: object.aria,
-  class: object.classname) do %>
+  class: object.classname + object.inline_class + object.compact_class) do %>
```
Comment (thestephenmarshall) [inline playbook/app/pb_kits/playbook/pb_select/select.html.erb:4]:
```suggestion
  class: object.classnames) do %>
```

File: playbook/app/pb_kits/playbook/pb_select/select.rb [hunk 2]
```diff
@@ -20,6 +22,14 @@ def classname
...
       end
 
+      def inline_class
+        inline ? " inline " : " "
+      end
+
+      def compact_class
+        compact ? "compact" : ""
+      end
+
```
Comment (thestephenmarshall) [inline playbook/app/pb_kits/playbook/pb_select/select.rb:32]:
```suggestion
def classnames
  classname + inline_class + compact_class
end

```

File: playbook/app/pb_kits/playbook/pb_select/select.html.erb (hunk 1 above)
Comment (thestephenmarshall) [reply playbook/app/pb_kits/playbook/pb_select/select.html.erb:4]: We could keep the template a little simpler by moving this into a method.

File: playbook/app/pb_kits/playbook/pb_select/select.rb (hunk 2 above)
Comment (thestephenmarshall) [reply playbook/app/pb_kits/playbook/pb_select/select.rb:32]: Method to create class

**Review by thestephenmarshall — COMMENTED** [review]
Looks good. Only one minor cleanup suggestion and then I did notice that the examples [here](https://pr1571.playbook.beta.hq.powerapp.cloud/kits/select/react) for React do not hold state when you make a selection.

**Comment by haydenrou** [comment]
@evalouderback do we want to use `angle-down` just for the compact version, or both inline versions? I just went with the kit default which was `chevron-down`.

**Comment by evalouderback** [comment]
@haydenrou I would do both inline versions

**Review by thestephenmarshall — APPROVED** [review]

**Comment by haydenrou** [comment]
I've changed the arrow and removed the forced uppercase.

**Comment by jasoncypret** [comment]
@bh247484 I'd like to discuss this PR with @kre8sions @evalouderback . I thought this wasn't being added to the design system???

**Comment by jasperfurniss** [comment]
@haydenrou Can you please add support for dark mode as well?

**Comment by haydenrou** [comment]
Hi @jasoncypret @bh247484
As far as BOP are concerned, we'd like to get this PR out in order to unblock 3 of our waiting stories, however there is still one outstanding question with the Icon part of the Backlog Item Type selection.

Brendan mentioned it was a very hacky solution in the handoff, and from discussions with Eva it seems at the moment that this is not going to be used all too often. My question is do we want that design to be a bespoke solution built into Runway based off of [this](https://pr234.playgrounds.beta.hq.powerapp.cloud/) prototype, or do we want the option of setting display icons to be passed into this kit. This will be an upcoming story but we'd like to know where that work will be based.

On top of that, I'd be interested whether after having the discussion of this prototype, was there by any chance another solution discussed that may be cleaner and add less technical debt? cc @thestephenmarshall 
(at the moment the `<select>` is transparent and there is a `div` over the top of it to display the icon)

Cheers

**Review by bh247484 — APPROVED** [review]

**Comment by jasoncypret** [comment]
@haydenrou I just saw your question. We might want to chat. I requested this be built differently from Kevin and was assuming we would go down a different route all together. Happy to share our ideas and get yours as well, but the quick solution would be to build a dropdown kit. It's built to do these kind of things.

CC @evalouderback @thestephenmarshall

**Comment by haydenrou** [comment]
I believe Brendan and Eva have talked and came up with a changed design solution that will make this a lot easier for us. Notes are [here](https://nitro.powerhrg.com/runway/backlog_items/BOP-1031)

**Pull request merged**